### PR TITLE
Refactor chain_get_block method

### PR DESCRIPTION
**Pull request by ZhmakAS**
* Resolves: # 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits are signed
- [ ] Tests included/updated or not needed
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: casper/types.go [hunk 1]
```diff
@@ -11,9 +11,9 @@ type (
 	Args            = types.Args
 	ValidatorBid    = types.ValidatorBid
 	AuctionBid      = types.AuctionBid
```
Comment (davidatwhiletrue) [inline casper/types.go:13]: Should this keep the original types?

**Review by davidatwhiletrue — APPROVED** [review]
@ZhmakAS see my comments.

File: types/block.go [hunk 2]
```diff
@@ -5,15 +5,23 @@ import (
...
+	Block  Block   `json:"block"`
+	Proofs []Proof `json:"proofs"`
+}
+
 type Block struct {
-	Hash   key.Hash    `json:"hash"`
-	Header BlockHeader `json:"header"`
-	Body   BlockBody   `json:"body"`
-	Proofs []Proof     `json:"proofs"`
+	BlockV1 *BlockV1 `json:"Version1"`
```
Comment (koltsov-iv) [inline types/block.go:14]: If we do so, it breaks all clients using the Block structure, father more the fields of this structure contains also different types, so clients should be rewritten significantly.
```
var eraID uint32
if (s.switchBlock.BlockV1 != nil) {
eraID = s.switchBlock.BlockV1.Header.EraID 
} else {
eraID = s.switchBlock.BlockV1.Header.EraID
}
```
and so on for each property....
Don't we want to come up with other ideas that will save customers from this pain?

**Review by koltsov-iv — COMMENTED** [review]
Please check the comment

File: types/block.go (hunk 2 above)
Comment (koltsov-iv) [reply types/block.go:14]: Are you sure that block version is important for most cases? I'm worrying about the smooth transfer of SDK clients to v0.2. With the current design when v2.0 is released, they can't use it with the old SDK version as well and they won't easily updates to the new version without significant code changes. The process should take dev time, losing data and following migration problems. I suggest thinking about a possibility to simplify this process for clients.

Comment (davidatwhiletrue) [reply types/block.go:14]: I took the same approach for .NET SDK. And, as far as we know, Casper took also the same decision with Javascript SDK. SDKs for v1.x are not working on v2.x, and viceversa.

File: sse/event.go [hunk 3]
```diff
@@ -52,8 +52,8 @@ type APIVersionEvent struct {
 // BlockAddedEvent definition
 type (
 	BlockAdded struct {
-		BlockHash string      `json:"block_hash"`
-		Block     types.Block `json:"block"`
+		BlockHash string        `json:"block_hash"`
+		Block     types.BlockV1 `json:"block"`
```
Comment (koltsov-iv) [inline sse/event.go:56]: Should it be BlockWrapper? Or how to deal with v2?

**Review by koltsov-iv — APPROVED** [review]

File: rpc/response.go [hunk 4]
```diff
@@ -197,3 +222,62 @@ type InfoGetChainspecResult struct {
...
+	}
+	if err := json.Unmarshal(bytes, &result); err != nil {
+		return err
+	}
+
+	if result.BlockHeader.BlockHeaderV2 == nil {
+		return errors.New("incorrect RPC response structure")
+	}
+
+	*h = QueryGlobalStateResult{
```
Comment (Volodymyr-Kuchinskyi) [inline rpc/response.go:274]: is it safe to dereference pointer here ?
I remember we got some panics with such an approach before, isn't it safer to do this way ?
```golang
value := QueryGlobalStateResult{}
h := &value
```

**Pull request merged**